config/clients/js/template/client.mustache

```diff
@@ -0,0 +1,438 @@
+    await chunkSequentialCall<TupleKey, void>(
+      (chunk) => this.api.write(
+        { writes: { tuple_keys: chunk}, authorization_model_id: authorizationModelId },
+        { retryParams: { maxRetry: DEFAULT_MAX_RETRY_OVERRIDE }, headers })
+        .then(() => { results.writes.push(...chunk.map(tuple => ({ tuple_key: tuple, status: ClientWriteStatus.SUCCESS }))); })
+        .catch((err) => { results.writes.push(...chunk.map(tuple => ({ tuple_key: tuple, status: ClientWriteStatus.FAILURE, err }))); }),
+      writes || [],
+      maxPerChunk,
+    );
+    await chunkSequentialCall<TupleKey, void>(
+      (chunk) => this.api.write(
+        { deletes: { tuple_keys: chunk }, authorization_model_id: authorizationModelId },
+        { retryParams: { maxRetry: DEFAULT_MAX_RETRY_OVERRIDE }, headers })
+        .then(() => { results.deletes.push(...chunk.map(tuple => ({ tuple_key: tuple, status: ClientWriteStatus.SUCCESS }))); })
+        .catch((err) => { results.deletes.push(...chunk.map(tuple => ({ tuple_key: tuple, status: ClientWriteStatus.FAILURE, err }))); }),
+      deletes || [],
+      maxPerChunk,
+    );
+
+    return results;
+  }
+
+  /**
+   * WriteTuples - Utility method to write tuples, wraps Write
+   * @param tuples
+   * @param options
+   */
+  async writeTuples(tuples: TupleKey[], options: ClientRequestOptsWithAuthZModelId & ClientWriteRequestOpts = {}): Promise<ClientWriteResponse> {
+    const { headers = {} } = options;
+    setHeaderIfNotSet(headers, CLIENT_METHOD_HEADER, "WriteTuples");
+    return this.write({ writes: tuples }, { ...options, headers });
+  }
+
+  /**
+   * DeleteTuples - Utility method to delete tuples, wraps Write
+   * @param tuples
+   * @param options
+   */
+  async deleteTuples(tuples: TupleKey[], options: ClientRequestOptsWithAuthZModelId & ClientWriteRequestOpts = {}): Promise<ClientWriteResponse> {
+    const { headers = {} } = options;
+    setHeaderIfNotSet(headers, CLIENT_METHOD_HEADER, "DeleteTuples");
+    return this.write({ deletes: tuples }, { ...options, headers });
+  }
+
+  /************************
+   * Relationship Queries *
+   ************************/
+
+  /**
+   * Check - Check if a user has a particular relation with an object (evaluates)
+   * @param checkRequest
+   * @param options
+   */
+  async check(checkRequest: ClientCheckRequest, options: ClientRequestOptsWithAuthZModelId = {}): PromiseResult<CheckResponse> {
+    return this.api.check({
+      tuple_key: {
+        user: checkRequest.user,
+        relation: checkRequest.relation,
+        object: checkRequest.object,
+      },
+      contextual_tuples: { tuple_keys: checkRequest.contextual_tuples || [] },
+      authorization_model_id: this.getAuthorizationModelId(options)
+    }, options).then(response => ({
+      ...response,
+      allowed: response.allowed || false,
+    }));
+  }
+
+  /**
+   * BatchCheck - Run a set of checks (evaluates)
+   * @param batchCheckRequest
+   * @param options
```

Comment:
similarly to above - perhaps document maxParallelRequests here so that caller does not need to read inside the method

Reply:
Done